### PR TITLE
ARROW-17732: [Docs][Java] Add minimal JDBC driver docs

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Comment by lidavidm** [comment]
Accompanies https://github.com/apache/arrow-site/pull/236

Rendered:

image

**Comment by nealrichardson** [comment]
@lidavidm I think ARROW-17332 is not the right jira

**Comment by lidavidm** [comment]
CC @jduo, does this look reasonable as a basic landing page?

**Comment by lidavidm** [comment]
@davisusanibar @lwhite1 could one of you give this a skim? It would be good to have documentation for our shiny new driver.

File: docs/source/java/flight_sql_jdbc_driver.rst [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+       </dependency>
+     </dependencies>
+   </project>
+
+Connecting to a Database
+========================
+
+The URI format is as follows::
+
+  jdbc:arrow-flight-sql://HOSTNAME:PORT[/?param1=val1&param2=val2&...]
```
Comment (lwhite1) [inline docs/source/java/flight_sql_jdbc_driver.rst:67]: Probably just me but I find these URI format definitions to be a little cryptic. Maybe include an example string with a couple of parameters so people have something concrete to refer to?

**Review by lwhite1 — COMMENTED** [review]
One suggestion. Otherwise LGTM.

File: docs/source/java/flight_sql_jdbc_driver.rst (hunk 1 above)
Comment (lidavidm) [reply docs/source/java/flight_sql_jdbc_driver.rst:67]: Thanks, good idea. I added an example with some prose to explain the parameters.

**Pull request merged**